library/src/main/java/com/getstream/sdk/chat/rest/core/providers/StreamWebSocketServiceProvider.java

```diff
@@ -32,13 +32,30 @@ public StreamWebSocketServiceProvider(ApiClientOptions options, String apiKey) {
     }
 
     @Override
-    public WebSocketService provideWebSocketService(User user, String userToken, WSResponseHandler listener) throws UnsupportedEncodingException {
+    public WebSocketService provideWebSocketService(User user, @Nullable String userToken, WSResponseHandler listener) throws UnsupportedEncodingException {
         String wsUrl = getWsUrl(userToken, user);
         Log.d(TAG, "WebSocket URL : " + wsUrl);
         return new StreamWebSocketService(wsUrl, listener);
     }
 
-    @NotNull
+    public String getWsUrl(@Nullable String userToken, User user) throws UnsupportedEncodingException {
+        String json = buildUserDetailJSON(user);
+
+        try {
+            json = URLEncoder.encode(json, StandardCharsets.UTF_8.toString());
+            String baseWsUrl = apiClientOptions.getWssURL() + "connect?json=" + json + "&api_key=" + apiKey;
+            if (userToken != null) {
```

Comment:
better to add `anonymousAuth` to both `getWsUrl` and `provideWebSocketService`; this way code is much easier to read + you can raise an error if `anonymousAuth == false && userToken == null`

this is a breaking change but for pretty internal stuff (and our tests)